server/src/main/java/io/pravega/schemaregistry/storage/impl/group/Group.java

```diff
@@ -0,0 +1,701 @@
+import static io.pravega.schemaregistry.storage.impl.group.records.TableRecords.EncodingInfoRecord;
+import static io.pravega.schemaregistry.storage.impl.group.records.TableRecords.GroupPropertiesRecord;
+import static io.pravega.schemaregistry.storage.impl.group.records.TableRecords.CodecTypesListValue;
+import static io.pravega.schemaregistry.storage.impl.group.records.TableRecords.LatestEncodingIdKey;
+import static io.pravega.schemaregistry.storage.impl.group.records.TableRecords.SchemaFingerprintKey;
+import static io.pravega.schemaregistry.storage.impl.group.records.TableRecords.GroupPropertyKey;
+import static io.pravega.schemaregistry.storage.impl.group.records.TableRecords.IndexTypeVersionToIdKey;
+import static io.pravega.schemaregistry.storage.impl.group.records.TableRecords.SchemaVersionList;
+import static io.pravega.schemaregistry.storage.impl.group.records.TableRecords.CodecTypesKey;
+import static io.pravega.schemaregistry.storage.impl.group.records.TableRecords.ValidationPolicyKey;
+import static io.pravega.schemaregistry.storage.impl.group.records.TableRecords.LatestEncodingIdValue;
+import static io.pravega.schemaregistry.storage.impl.group.records.TableRecords.SchemaIdValue;
+
+/**
+ * Class that implements all storage business logic for a group.
+ * The group uses a key value store which supports batch updates with optimistic concurrency.
+ *
+ * @param <V> Type of version used in the index.
+ */
+@Slf4j
+public class Group<V> {
+    private static final TableRecords.Etag ETAG = new TableRecords.Etag();
+    private static final ValidationPolicyKey VALIDATION_POLICY_KEY = new ValidationPolicyKey();
+    private static final GroupPropertyKey GROUP_PROPERTY_KEY = new GroupPropertyKey();
+    private static final CodecTypesKey CODECS_TYPE_KEY = new CodecTypesKey();
+    private static final LatestSchemasKey LATEST_SCHEMAS_KEY = new LatestSchemasKey();
+    private static final LatestEncodingIdKey LATEST_ENCODING_ID_KEY = new LatestEncodingIdKey();
+    private static final Retry.RetryAndThrowConditionally WRITE_CONFLICT_RETRY = 
+            Retry.withExpBackoff(1, 2, Integer.MAX_VALUE, 100)
+                 .retryWhen(x -> Exceptions.unwrap(x) instanceof StoreExceptions.WriteConflictException);
+    private static final CodecType NONE_CODEC_TYPE = new CodecType("");
+
+    private final GroupTable<V> groupTable;
+    private final ScheduledExecutorService executor;
+
+    public Group(GroupTable<V> groupTable, ScheduledExecutorService executor) {
+        this.groupTable = groupTable;
+        this.executor = executor;
+    }
+
+    public CompletableFuture<Boolean> create(SerializationFormat serializationFormat, ImmutableMap<String, String> properties,
+                                          boolean allowMultipleTypes, Compatibility compatibility) {
+        List<Map.Entry<TableKey, GroupTable.Value<TableValue, V>>> entries = new ArrayList<>();
```

Comment:
It's quite a complicated structure. A comment would help a great deal for readability purposes.